openslides_backend/action/action_handler.py

```diff
@@ -88,22 +92,24 @@ def handle_request(self, payload: Payload, user_id: int) -> ActionResponse:
         retried = 0
         payload_copy = deepcopy(payload)
         while True:
-            # Parse actions and creates events
-            write_request, results = self.parse_actions(payload)
-
-            # Send events to datastore
-            if write_request:
-                try:
+            try:
+                # Parse actions and creates events
+                write_request, results = self.parse_actions(payload)
+                if write_request:
+                    # Send events to datastore
                     self.datastore.write(write_request)
-                except DatastoreException as exception:
-                    retried += 1
-                    payload = deepcopy(payload_copy)
-                    if retried > self.MAX_RETRY:
-                        raise ActionException(exception.message)
-                    continue
-                except EventStoreException as exception:
+            except DatastoreLockedException as exception:
+                retried += 1
+                payload = deepcopy(payload_copy)
+                if retried > self.MAX_RETRY:
                     raise ActionException(exception.message)
+                continue
+            except ServiceException as exception:
+                raise ActionException(exception.message)
+            except Exception as exception:
```

Comment:
Did you had this catch-all in there previously? I can understand ist, but this is too high in the structure. This catch-all should be done in the http request handling method as far down as possible, so it really catches unwanted errors.

Here, you can be more specific. What exceptions do you expect to catch here? Please explicitly name and handle them. I would also propose splitting this method if possible to seperate error handling from other logic.

Reply:
Respecting the catch-all I agree with you and removed it.
But for the rest we have only 2 kind of exceptions:
1. Special ones, we want to retry the action. This kind has to be a special expression type and has to be listed together with the DatastoreLockedException as a tuple. If we need another reaction, we would catch the specific Exception-classes in another case.
2. The rest of our foreseeable Exception are the ServiceException, which should be the base class of all services raising exceptions. They are all converted to ActionExceptions, that is the common reaction for all of them.

I included the parse_actions-method into the try-block, because there could be at minimum a MediaServiceException, which is also a ServiceException with the same handling.